logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/ProtoContentMapper.kt

```diff
@@ -49,11 +60,23 @@ class ProtoContentMapperImpl: ProtoContentMapper {
             is GenericMessage.Content.Composite -> MessageContent.Unknown
             is GenericMessage.Content.Confirmation -> MessageContent.Unknown
             is GenericMessage.Content.DataTransfer -> MessageContent.Unknown
-            is GenericMessage.Content.Deleted -> MessageContent.Unknown
+            is GenericMessage.Content.Deleted -> MessageContent.DeleteMessage(genericMessage.messageId)
             is GenericMessage.Content.Edited -> MessageContent.Unknown
             is GenericMessage.Content.Ephemeral -> MessageContent.Unknown
             is GenericMessage.Content.External -> MessageContent.Unknown
-            is GenericMessage.Content.Hidden -> MessageContent.Unknown
+            is GenericMessage.Content.Hidden -> {
+                val hiddenMessage = genericMessage.hidden
+                if (hiddenMessage != null) {
+                    MessageContent.DeleteForMe(
+                        hiddenMessage.messageId,
+                        ConversationId(hiddenMessage.qualifiedConversationId?.id!!,
```

Comment:
Let's unwrap and return `MessageContent.Unknown` unknown if the `qualifiedConversationId` is null otherwise we will crash on bad data.

or maybe even better

we assume that the domain is the domain of the self user if it's not specified.

Reply:
Forced it to crash. because the important part is the message hidden content, if it's null then it deserve a crash :D

Reply:
No, we can't crash here because then a malicious person or simply a user with an old client can crash the android client at will by sending a bad delete message.